byron/ledger/executable-spec/src/Ledger/Core.hs

```diff
@@ -151,34 +156,34 @@ minusSlotMaybe (Slot m) (SlotCount n)
   | otherwise = Just . Slot $ m - n
 
 newtype BlockCount = BlockCount { unBlockCount :: Word64 }
-  deriving (Eq, Generic, Ord, Num, Show)
+  deriving (Eq, Generic, Ord, Show)
+  deriving newtype (Num, Hashable)
 
 instance HasTypeReps BlockCount
 
+
 ---------------------------------------------------------------------------------
 -- Transactions
 ---------------------------------------------------------------------------------
 
 -- |The address of a transaction output, used to identify the owner.
 newtype Addr = Addr VKey
-  deriving (Show, Eq, Ord, HasOwner)
+  deriving (Show, Eq, Ord, Generic)
```

Comment:
Some of these things have the potential to be a little confusing, since e.g. `Eq` and `Ord` will also both be derived via newtype. Given the default behaviour changes with `DeriveAnyClass` enabled, there's some potentially hidden subtlety here. It might be nicer to split these instead into `deriving stock (Show, Generic)` and move `Eq` and `Ord` into the newtype deriving section to remove ambiguity?

Reply:
@nc6 That sounds like a good idea - are show and generic the only ones that default to stock that we're using here?

Reply:
I think so. https://gitlab.haskell.org/ghc/ghc/wikis/commentary/compiler/deriving-strategies#the-deriving-strategy-resolution-algorithm goes through in detail